src/controllers/confirmation.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { NextFunction, Request, Response } from "express";
+import { PTFCompanyProfile } from "../model/company.profile";
+import { CONFIRMATION_NOT_REQUIRED, CONFIRMATION_STILL_REQUIRED } from "../model/template.paths";
+import * as sessionService from "../services/session.service";
+import {COMPANY_PROFILE, USER_PROFILE} from "../session/keys";
+import {IUserProfile} from "../session/types";
+
+const createMissingError = (item: string): Error => {
+    const errMsg: string = item + " missing from session";
+    return new Error(errMsg);
+};
+
+const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  const companyProfile: PTFCompanyProfile =
+      await sessionService.getPromiseToFileSessionValue(req.chSession, COMPANY_PROFILE);
+
+  if (!companyProfile) {
+    return next(createMissingError("Company profile"));
+  }
+
+  const signInInfo = req.chSession.getSignedInInfo();
+  const userProfile: IUserProfile = signInInfo[USER_PROFILE] as IUserProfile;
+  const email = userProfile.email;
+
+  if (!email) {
+    return next(createMissingError("User Email"));
+  }
+  // TODO LFA-1169 Clarify what the ext deadline period will be for accounts and cs.
+
+  // TODO LFA-TBC call promise-to-file api
+
+  // TODO LFA-1380 need to add decision "is company still required" as a flag to the session.
+  const isRequired: boolean = false;
+  if (isRequired) {
+    return res.render(CONFIRMATION_STILL_REQUIRED,
+     {
+       company: companyProfile,
+       userEmail: email,
+     });
+  } else {
+    return res.render(CONFIRMATION_NOT_REQUIRED,
```

Comment:
Sorry, should have spotted this earlier, but shouldn't this constant be 'COMPANY_NOT_REQUIRED' rather than 'CONFIRMATION...'?

Also, wondering why there's code added for the 'still required' scenario because that's not the screen related to this story - https://companieshouse.atlassian.net/browse/LFA-1166. Guessing that's a later requirement/story?

Reply:
No it refers to the comfirmation screen we need soemthingt hat corresponds to the url name.

Reply:
Ah I see (company is kind of implied here). 👍

Reply:
Ok, understand about the company/confirmation now but I still wonder why this PR is doing stuff related to the 'still required' scenario?